test/src/org/labkey/test/tests/targetedms/TargetedMSChromatogramOptimizationTest.java

```diff
@@ -0,0 +1,133 @@
+        init.setupFolder(FolderType.Library);
+    }
+
+    @Override
+    protected String getProjectName()
+    {
+        return "TargetedMS Chromatogram Optimization Test";
+    }
+
+    @Test
+    public void testUpload() throws Exception
+    {
+        goToProjectHome();
+        importData(SKY_FILE);
+        clickAndWait(Locator.linkContainingText("Panorama Dashboard"));
+        File downloadedClibFile = doAndWaitForDownload(() -> clickButton("Download", 0));
+
+        goToProjectHome();
+        String queryURL = "query-executeQuery.view?schemaName=targetedms&query.queryName=TransitionOptimization";
+        goToURL(new URL(WebTestHelper.getBaseURL() + getCurrentContainerPath() + "/" + queryURL), 60000);
+        DataRegionTable table = new DataRegionTable("query",getDriver());
+        table.rowSelector().showAll();
+
+        checker().verifyEquals("Invalid number of rows in transition optimization", table.getDataRowCount(),
+                sizeOfTable(downloadedClibFile, "MoleculeTransitionOptimization"));
+
+        log("Verifying the new tables added");
+        checker().verifyTrue("TransitionOptimization table is not present in the SQLITE file",
+                tableExists(downloadedClibFile, "TransitionOptimization"));
+        checker().verifyTrue("MoleculeTransitionOptimization table is not present in SQLITE file",
+                tableExists(downloadedClibFile, "MoleculeTransitionOptimization"));
+
+        log("Verifying the SampleFile modifications");
+        checker().verifyTrue("Sample File does not have CePredictorId",
+                columnExists(downloadedClibFile,"SampleFile","CePredictorId"));
+        checker().verifyTrue("Sample File does not have DpPredictorId",
+                columnExists(downloadedClibFile,"SampleFile","DpPredictorId"));
+
+    }
+
+    private int sizeOfTable(File clibFile, String name)
+
+    {
+        int cnt = 0;
+        @SuppressWarnings("SqlResolve")
+        String sql = "SELECT * FROM " + name;
+        ConnectionSource cs = new ConnectionSource(clibFile.getAbsolutePath());
+        Connection conn = null;
+        try
+        {
+            conn = cs.getConnection();
+            ResultSet rs = conn.createStatement().executeQuery(sql);
+            while (rs.next())
+                cnt++;
+        }
+        catch (SQLException e)
+        {
+            e.printStackTrace();
+        }
+        return cnt;
+    }
+
+    private boolean tableExists(File clibFile, String tableName)
+    {
+        ConnectionSource cs = new ConnectionSource(clibFile.getAbsolutePath());
+        try
+        {
+            Connection conn = cs.getConnection();
+            DatabaseMetaData md = conn.getMetaData();
+            ResultSet rs = md.getTables(null, null, tableName, null);
+            if (rs.next())
+                return true; //Table exists
+            else
+                return false;
+        }
+        catch (SQLException e)
+        {
+            e.printStackTrace();
```

Comment:
This exception shouldn't be caught and swallowed like this. I recommend just letting it bubble up through the call stack

Reply:
Done